roaring/roaring_internal_test.go

```diff
@@ -2410,3 +2416,83 @@ func TestSearc64(t *testing.T) {
 		})
 	}
 }
+
+func TestIntersectArrayBitmap(t *testing.T) {
+	a, b := &container{}, &container{
+		bitmap: make([]uint64, bitmapN),
+	}
+	tests := []struct {
+		array  []uint16
+		bitmap []uint64
+		exp    []uint16
+	}{
+		{
+			array:  []uint16{0},
+			bitmap: []uint64{1},
+			exp:    []uint16{0},
+		},
+		{
+			array:  []uint16{0, 1},
+			bitmap: []uint64{3},
+			exp:    []uint16{0, 1},
+		},
+		{
+			array:  []uint16{64, 128, 129, 2000},
+			bitmap: []uint64{932421, 2},
+			exp:    []uint16{},
+		},
+		{
+			array:  []uint16{0, 65, 130, 195},
+			bitmap: []uint64{255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255},
+			exp:    []uint16{0, 65, 130, 195},
+		},
+		{
+			array:  []uint16{63, 120, 543, 639, 12000},
+			bitmap: []uint64{0x8000000000000000, 0, 0, 0, 0, 0, 0, 0, 0, 0x8000000000000000},
+			exp:    []uint16{63, 639},
+		},
+		{
+			array:  []uint16{0, 1, 63, 120, 543, 639, 12000, 65534, 65535},
+			bitmap: bitmapOdds(),
+			exp:    []uint16{1, 63, 543, 639, 65535},
+		},
+		{
+			array:  []uint16{0, 1, 63, 120, 543, 639, 12000, 65534, 65535},
+			bitmap: bitmapEvens(),
+			exp:    []uint16{0, 120, 12000, 65534},
+		},
+	}
+
+	for i, test := range tests {
+		a.array = test.array
+		a.container_type = ContainerArray
+		for i, bmval := range test.bitmap {
```

Comment:
we should use helper functions to create full bitmaps and get rid of this gross for loop